### DISCUSSION TITLE
Planning: minimum Python 3.7

### DISCUSSION THREAD

**Discussion by scivision** [mesonbuild/meson#9781] [Ideas]
By PyPI downloads, Python 3.8 surpassed Python 3.6 Meson downloads ~ November 2021. Minimum supported Python 3.7 planning as Python 3.6 falls out of use/support. Using similar criteria to drop Python < 3.6: [#6297](https://github.com/mesonbuild/meson/issues/6297)

![meson202201](https://user-images.githubusercontent.com/10931741/148335286-9d8115c6-7569-459d-95a1-c4e3eb642ecb.png)

* Ubuntu 20.04 Python 3.8 
* Ubuntu 18.04 (EOL standard support April 2023) Python 3.6
* Debian stable Python 3.9
* RHEL 8 AppStream Python 3.9

Benefits include: 
* Python 3.10 type annotation improvements available in Python >= 3.7 via `from __future__ import annotations`
* asyncio improvements

**Comment by eli-schwartz** [comment]
I have a PR for this: #9510

**Reply by scivision** [reply]
Thanks! I searched Meson issue but not PRs 😨

**Comment by eli-schwartz** [comment]
Add to the list of improvements:
- dataclasses
  Quoting @dcbaker
  > dataclasses I find generally help encourage writing better classes that don't do a bajillion things in their initializers.
- importlib.resources to replace embedded files as python strings (mesondata.py)

Add to the list of supported python versions:
- Debian oldstable Python 3.7
